net/pfSense-pkg-cellular/files/usr/local/www/widgets/widgets/cellular.widget.php

```diff
@@ -0,0 +1,237 @@
+$python_bin = '/usr/local/bin/python2.7';
+$interface_bin = '/usr/local/sbin/cellular';
+$widget_url = '/widgets/widgets/cellular.widget.php';
+
+if (isset($_GET['getsignalstrength'])):
+
+    if (file_exists($python_bin) && file_exists($interface_bin)) {
+        exec("$python_bin $interface_bin signal", $sig);
+        print $sig[0];
+    };
+
+elseif (isset($_GET['getcarrier'])):
+
+    if (file_exists($python_bin) && file_exists($interface_bin)) {
+        exec("$python_bin $interface_bin carrier", $carrier);
+        print $carrier[0];
+    };
+
+elseif (isset($_GET['getinfoex'])):
+
+    if (file_exists($python_bin) && file_exists($interface_bin)) {
+        exec("$python_bin $interface_bin infoex", $infoex);
+        print $infoex[0];
+    };
+
+elseif (isset($_GET['widget'])):
+    if (file_exists($python_bin) && file_exists($interface_bin)) {
+        exec("$python_bin $interface_bin widget", $widget);
+        print $widget[0];
+        // print $widget[1];
+    };
+
+elseif (isset($_GET['getstatus'])):
+        $ifdescrs = get_configured_interface_with_descr();
+        foreach ($ifdescrs as $ifdescr => $ifname) {
+            $ifinfo = get_interface_info($ifdescr);
+            if ($ifinfo['ppplink']) {
+                $known_status = true;
+                $icon = "";
+
+                // Choose an icon by interface status
+                if ($ifinfo['status'] == "up" || $ifinfo['status'] == "associated") {
+                    $icon = 'arrow-up text-success';
+                } elseif ($ifinfo['status'] == "no carrier") {
+                    $icon = 'times-circle text-danger';
+                } elseif ($ifinfo['status'] == "down") {
+                    $icon = 'arrow-down text-danger';
+                } else {
+                    $known_status = false;
+                }
+                $macaddr = $ifinfo['macaddr'];
+                $status = $ifinfo['status'];
+                $ipaddr = htmlspecialchars($ifinfo['ipaddr']);
+                $ipaddrv6 = htmlspecialchars($ifinfo['ipaddrv6']);
+                $ifname = htmlspecialchars($ifname);
+                print "$known_status,$icon,$macaddr,$ifdescr,$ifname,$status,$ipaddr,$ipaddrv6";
+
+                break;
+            }
+
+        };
+
+else:
+?>
+
+<table id="modem" class="table">
+    <tbody>
+    <tr>
+    <td id="macaddr">
+    <div id="signal-strength"><div id="signal-strength-bars"></div></div>
+    <a id="ifname"></a></td>
+    <td id="status"></td>
+    <td id="carrier"><span id="carrier-span"></span>
+        <strong id="modem-mode" class="icn"></strong></td>
+    <td id="ipaddr"></td>
+    </tr>
+    </tbody>
+</table>
+
+<script type="text/javascript">
+    function widget() {
+        var url = <?= '"' . $widget_url . '"' ?>;
+        var pars = 'widget';
+        var myAjax = $.ajax({
+                url: url,
+                type: "get",
+                data: pars,
+                complete: widgetcallback_modem
+            });
+    }
+
```

Comment:
We are trying to convert pfSense to POST operation wherever possible. Perhaps it would be best to start of that way here?